### DISCUSSION TITLE
Mock service by Kuksa VAL

### DISCUSSION THREAD

**Discussion by antaradey-itk** [eclipse-leda/leda-distro#154] [Q&A] [answered]
Hi,

I have developed a small application using the Mockservice which can be used to mock datapoints using a configuration file. The service is provided by the Kuksa VAL as below: 
https://github.com/eclipse/kuksa.val.services/tree/main/mock_service

Another way is to develop the application is to directly use the Kuksa Client library to trigger the actutators. 

The docker image for the Mockservice is 364MB. When deploying it to the Leda device, it takes a very long time (~10 mins), and eventually fails to run due to low disk space.

Similar problem also happened while deploying a different large sized image to a Raspberry Pi device.

My question is: 
Is the Leda image not equipped to handle large container sizes? What is the goal/roadmap here?

**Comment by vasilvas99** [comment]
Hi, 
The slow download on qemu can be due to the user-level networking (slirp) used, using virtual tun/tap devices is a bit more complicated to set-up, so let's take a look at the out of space issues first.

Can you give me the output of the command `df -h`?

**Reply by antaradey-itk** [reply]
[before_Mock_df.txt](https://github.com/eclipse-leda/leda-distro/files/12683082/before_Mock_df.txt)

This is the output of df -h before any other custom application is deployed in Leda.

**Reply by vasilvas99** [reply]
Hey @antaradey-itk , sorry for the late response. Those before logs look normal.
Can you give me similar outputs of `df` after you deploy you app and after you run out of space.

What I'd like to see is the data partition getting filled-up as it's where such data is supposed to go and can be easily expanded.  If one of the root partitions runs out of space, it might mean that there is a problem with the containerd configuration.

Thanks for assistance in debugging this problem :)

**Reply by antaradey-itk** [reply, marked as answer]
Hi @vasilvas99, 

I was on vacation the previous month, so I could not respond to the thread. I had done a clean up before leaving, and created a fresh Leda environment. The previous output file of df -h was created from the new workspace. 
I have deployed the mock service again in this environment again, and I don't see the memory issue anymore, and also runs smoothly. It is a strange behavior, as the issue was persistent earlier. I shall keep you updated if I face this issue again, or with any other container images. 

Thank you for your support. 💯
